react/features/dialog/components/Dialog.js

```diff
@@ -0,0 +1,154 @@
+/* global APP */
+
+import ModalDialog from '@atlaskit/modal-dialog';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { SpeakerStats } from '../../speaker-stats';
+import { hideDialog, showDialog } from '../actions';
+
+const dialogs = {
+    SpeakerStats
+};
+
+/**
+ * React component for dialog.
+ *
+ * @extends Component
+ */
+class Dialog extends Component {
+    /**
+     * Dialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        config: React.PropTypes.object,
+        dispatch: React.PropTypes.func
+    }
+
+    /**
+     * Initializes new Dialog component.
+     *
+     * @param {Object} props - Component props.
+     */
+    constructor(props) {
+        super(props);
+        this._onCloseDialog = this._onCloseDialog.bind(this);
+    }
+
+    /**
+     * Register key press listeners.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillMount() {
+        APP.keyboardshortcut.registerShortcut(
+          'T',
+          'showSpeakerStats',
+          this.toggleDialog.bind(this, 'SpeakerStats', 'Speaker Stats'),
+          'keyboardShortcuts.showSpeakerStats'
+        );
+    }
+
+    /**
+     * Remove registered key press listeners.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        APP.keyboardshortcut.unregisterShortcut('T');
+    }
+
+    /**
+     * Dispatches action to show a dialog of the given dialogType.
+     * The dialogType should match one that exists in "dialogs."
+     *
+     * @param {string} dialogType - Which dialog to open.
+     * @param {string} dialogTitle - Title of the dialog.
+     * @returns {void}
+     */
+    toggleDialog(dialogType, dialogTitle) {
+        if (this.props.config.dialogType === dialogType) {
+            this._onCloseDialog();
+        } else {
+            const action = showDialog(dialogType, {
+                title: dialogTitle
+            });
+
+            this.props.dispatch(action);
+        }
+    }
+
+    /**
+     * Dispatches action to hide the dialog.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onCloseDialog() {
+        this.props.dispatch(hideDialog());
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const config = this.props.config || {};
+        const DialogComponent = dialogs[config.dialogType];
+        const dialogConfig = config.options;
+
+        let dialog = null;
+
+        if (DialogComponent) {
```

Comment:
You can reduce one level of indentation if you return early. Just return null if there is no component.